### PR TITLE
feat: Describe App Storage tags API

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
### Description

The PR documents tags API that has been already existed in App Storage for a while.

### Motivation and Context

https://saucedev.atlassian.net/browse/LT-2996

### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/dev/api/storage.md [hunk 1]
```diff
@@ -659,6 +663,12 @@ Uploads an app file to Sauce Storage for the purpose of mobile app testing and r
      <td><p><small>| FORM-TEXT | OPTIONAL | STRING |</small></p><p>A description to distinguish your app.</p></td>
     </tr>
   </tbody>
+  <tbody>
+    <tr>
+     <td><code>tags</code></td>
+     <td><p><small>| FORM-TEXT | OPTIONAL | STRING |</small></p><p>An optional list of comma-separated tag names assigned to the uploaded file. Each tag name length must be between 1 and 16 characters. Tag names must only consist of latin uppercase (A-Z), latin lowercase (a-z), digits (0-9), underscore ("_"), hyphen ("-") and dot (".") characters. Tag names are case-sensitive. It is allowed to assign up to 10 tags to a single file.</p></td>
```
Comment (LoryApiFortress) [inline docs/dev/api/storage.md:675]:
```suggestion
     <td><p><small>| FORM-TEXT | OPTIONAL | STRING |</small></p><p>An optional list of comma-separated tag names assigned to the uploaded file. Each tag name length must be between 1 and 16 characters. Tag names must only consist of uppercase (A-Z), lowercase (a-z), digits (0-9), underscore ("_"), hyphen ("-"), and dot (".") characters. Tag names are case-sensitive. It is allowed to assign up to 10 tags to a single file.</p></td>
```

File: docs/dev/api/storage.md [hunk 2]
```diff
@@ -844,7 +855,13 @@ Adds or updates the `description` attribute of the specified file.
   <tbody>
     <tr>
      <td><code>description</code></td>
-     <td><p><small>| BODY | REQUIRED | STRING |</small></p><p>A description to more clearly distinguish the stored file within the Sauce Labs system.</p></td>
+     <td><p><small>| BODY | OPTIONAL | STRING |</small></p><p>A description to more clearly distinguish the stored file within the Sauce Labs system.</p></td>
```
Comment (LoryApiFortress) [inline docs/dev/api/storage.md:864]:
```suggestion
     <td><p><small>| BODY | OPTIONAL | STRING |</small></p><p>A description to more clearly distinguish the stored file in the Sauce Labs system.</p></td>
```

File: docs/dev/api/storage.md [hunk 3]
```diff
@@ -844,7 +855,13 @@ Adds or updates the `description` attribute of the specified file.
...
+    </tr>
+  </tbody>
+    <tbody>
+    <tr>
+     <td><code>tags</code></td>
+     <td><p><small>| BODY| OPTIONAL | STRING |</small></p><p>An optional list of comma-separated tag names assigned to the uploaded file. Each tag name length must be between 1 and 16 characters. Tag names must only consist of latin uppercase (A-Z), latin lowercase (a-z), digits (0-9), underscore ("_"), hyphen ("-") and dot (".") characters. Tag names are case-sensitive. It is allowed to assign up to 10 tags to a single file. The value overrides the previously set tags.</p></td>
```
Comment (LoryApiFortress) [inline docs/dev/api/storage.md:870]:
```suggestion
     <td><p><small>| BODY | OPTIONAL | STRING |</small></p><p>An optional list of comma-separated tag names assigned to the uploaded file. Each tag name length must be between 1 and 16 characters. Tag names must only consist of uppercase (A-Z), lowercase (a-z), digits (0-9), underscore ("_"), hyphen ("-"), and dot (".") characters. Tag names are case-sensitive. It is allowed to assign up to 10 tags to a single file. The value overrides the previously set tags.</p></td>
```

Comment (LoryApiFortress) [reply docs/dev/api/storage.md:870]: @mykola-mokhnach Should the sentence "The value overrides the previously set tags" be also for the "tags" on line 668?

File: docs/dev/api/storage.md [hunk 4]
```diff
@@ -1173,3 +1194,84 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
 </details>
 
 ---
+
+### List Tags
+
+<details><summary><span className="api get">GET</span> <code>/v1/storage/tags</code></summary>
```
Comment (LoryApiFortress) [inline docs/dev/api/storage.md:1206]:
```suggestion
<details><summary><span className="api get">GET</span><code>/v1/storage/tags</code></summary>
```

File: docs/dev/api/storage.md [hunk 5]
```diff
@@ -1173,3 +1194,84 @@ curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
...
+<p/>
+
+Returns the list of tags available for your team sorted alphabetically. Each tag name is only returned if it is assigned to at least one file.
```
Comment (LoryApiFortress) [inline docs/dev/api/storage.md:1209]:
```suggestion
Returns the list of tags available for your team sorted alphabetically. Each tag name is only returned if assigned to at least one file.
```

**Review by LoryApiFortress — COMMENTED** [review]
Thank you @mykola-mokhnach! Left some comments

File: docs/dev/api/storage.md (hunk 3 above)
Comment (mykola-mokhnach) [reply docs/dev/api/storage.md:870]: no, on file upload nothing is overridden. It is always a new item

**Review by LoryApiFortress — APPROVED** [review]

**Pull request merged**